agent/src/global-state/AgentGlobalState.ts

```diff
@@ -45,12 +45,10 @@ export class AgentGlobalState implements vscode.Memento {
     }
 
     public async reset(): Promise<void> {
-        if (this.db instanceof InMemoryDB) {
-            this.db.clear()
+        this.db.clear()
```

Comment:
```suggestion
        if (this.db instanceof InMemoryDB) {
            this.db.clear()

            // HACK(sqs): Force `localStorage` to fire a change event.
            await localStorage.delete('')
        }
```

Reverting this change fixed the test for me locally 🤔 

![image](https://github.com/user-attachments/assets/a7c68710-11a5-40ff-815a-80665fdfda6e)

Reply:
never mind - if i restart it could fail